pulp_rpm/app/tasks/publishing.py

```diff
@@ -354,10 +356,10 @@ def publish(
     )
     with tempfile.TemporaryDirectory(dir="."):
         with RpmPublication.create(repository_version) as publication:
-            publication.metadata_checksum_type = get_checksum_type("primary", checksum_types)
-            publication.package_checksum_type = (
-                checksum_types.get("package") or publication.metadata_checksum_type
-            )
+            checksum_type = get_checksum_type("primary", checksum_types)
```

Comment:
should this use 'general'?

Reply:
No, this code is correct.  If you look at the definition of `get_checksum_type()`, it does query "general" first.  The string "primary" is used when looking up the original metadata checksum type noted at sync time, if the user didn't specify one.  And if the repo wasn't recently synced and the user didn't provide a specific one, then it falls back to default.